atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -52,32 +52,41 @@ private CassandraTimestampBoundStore(CassandraTimestampStore cassandraTimestampS
     @Override
     public synchronized long getUpperLimit() {
         DebugLogger.logger.debug("[GET] Getting upper limit");
+        currentLimit = getBoundToStore();
+        DebugLogger.logger.info("[GET] Setting cached timestamp limit to {}.", currentLimit);
+        return currentLimit;
+    }
+
+    private long getBoundToStore() {
         Optional<Long> currentBound = cassandraTimestampStore.getUpperLimit();
-        long boundToStore;
         if (!currentBound.isPresent()) {
             DebugLogger.logger.info("[GET] Null result, setting timestamp limit to {}", INITIAL_VALUE);
-            cassandraTimestampStore.storeTimestampBound(null, INITIAL_VALUE);
-            boundToStore = INITIAL_VALUE;
-        } else {
-            boundToStore = currentBound.get();
+            try {
+                cassandraTimestampStore.storeTimestampBound(Optional.empty(), INITIAL_VALUE);
+            } catch (ConcurrentModificationException e) {
+                throw constructMultipleServiceError(e);
+            }
+            return INITIAL_VALUE;
```

Comment:
return can go inside the catch block. Also, this try block is repeated in storeUpperLimit below - could extract the whole thing, with a `from` and `to` (or better-named versions).